libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteLine.java

```diff
@@ -170,16 +177,43 @@
 
     // Waypoint attributes
     int originWaypointIcon = typedArray.getResourceId(
-      R.styleable.NavigationMapRoute_originWaypointIcon, R.drawable.ic_route_origin);
+            R.styleable.NavigationMapRoute_originWaypointIcon, R.drawable.ic_route_origin);
     int destinationWaypointIcon = typedArray.getResourceId(
-      R.styleable.NavigationMapRoute_destinationWaypointIcon, R.drawable.ic_route_destination);
+            R.styleable.NavigationMapRoute_destinationWaypointIcon, R.drawable.ic_route_destination);
     typedArray.recycle();
 
     Drawable originIcon = drawableProvider.retrieveDrawable(originWaypointIcon);
     Drawable destinationIcon = drawableProvider.retrieveDrawable(destinationWaypointIcon);
-    belowLayer = findRouteBelowLayerId(belowLayer, style);
 
-    initializeLayers(style, layerProvider, originIcon, destinationIcon, belowLayer);
+    boolean layerBelowExists = false;
+    if (belowLayer != null && !belowLayer.isEmpty()) {
+      for (Layer layer : style.getLayers()) {
+        if (layer.getId().equals(belowLayer)) {
+          layerBelowExists = true;
+          break;
+        }
+      }
+    }
+
+    String belowLayerCandidate = null;
+    if (!layerBelowExists) {
+      for (int i = style.getLayers().size() - 1; i >= 0; i--) {
+        Layer layer = style.getLayers().get(i);
```

Comment:
```suggestion
        String layerId = style.getLayers().get(i).getId();
```

and other needed changes to make it happen